src/main/java/io/jenkins/plugins/aws/global_configuration/CredentialsAwsGlobalConfiguration.java

```diff
@@ -248,11 +247,14 @@ public ListBoxModel doFillCredentialsIdItems() {
     }
 
     public FormValidation doCheckRegion(@QueryParameter String region) {
-        FormValidation ret = FormValidation.ok();
-        if (StringUtils.isNotBlank(region) && Regions.fromName(region) == null) {
-            ret = FormValidation.error("Region is not valid");
+        if (StringUtils.isNotBlank(region)) {
```

Comment:
I am not a fan of more than one return point in a method ;)

Reply:
And I am not a fan of mutable variables. :-) Must have spent too much time in Common LISP as a youth :shrug:

Reply:
FTR the patch here was about the main bug fixed in this PR: an IAE displayed for an invalid region name, apparently introduced by an unexplained code change during the refactoring, which broke the downstream test case.